src/components/asset-list/RecyclerAssetList2/core/StickyHeaders.tsx

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+import { sortBy } from 'lodash';
+import React, {
+  MutableRefObject,
+  ReactElement,
+  useCallback,
+  useContext,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { Animated as RNAnimated, ScrollView } from 'react-native';
+import Animated from 'react-native-reanimated';
+import { useRecyclerAssetListPosition } from './Contexts';
+
+const Context = React.createContext<
+  | {
+      scrollViewRef: MutableRefObject<ScrollView | undefined>;
+      interpolationsRanges: Record<string, { range: number[]; last: boolean }>;
+      setMeasures: (position: number, height: number, name: string) => void;
+    }
+  | undefined
+>(undefined);
+
+export { Context as StickyHeaderContext };
+
+export function StickyHeader({
+  name,
+  children,
+}: {
+  name: string;
+  children: React.ReactChildren;
+}) {
+  const context = useContext(Context);
+  if (!context) {
+    return children;
```

Comment:
Should this throw an error instead? Presumably if the context value is undefined, it would actually be a bug? Also then we wouldn't need to disable the `react-hooks/rules-of-hooks` lint rule.

Reply:
We can use this in showcase I think, so this is good